### DISCUSSION TITLE
Error with element attribute having "::" with value being an expression #408

### DISCUSSION THREAD

**Discussion by KulvSingh** [a-h/templ#409] [General]
hx-on::after-swap={ fmt.Sprintf("htmx.find('#D001').focus()") }

This htmx attribute having "::" with value being an expression is evaluating to be an error as following

cannot use fmt.Sprintf("htmx.find('#D001').focus()") (value of type string) as templ.ComponentScript value in variable declarationcompiler[IncompatibleAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#IncompatibleAssign)

**Comment by a-h** [comment]
HI @KulvSingh, attributes that expect a script must be either a constant string:

```
hx-on::after-swap="htmx.find('#D001').focus()"
```

Or must be a `templ.ComponentScript`:

```
script onAfterSwap(id string) {
  htmx.find(id).focus()
}

hx-on::after-swap={ onAfterSwap("#D001") }
```

This is because templ is trying to prevent you from accidentally introducing an XSS. If you have a Go string expression, it may have come from an untrusted source (i.e. user input). templ's script templates (example 2 above) manage the JSON escaping of Go values and pass them to a JavaScript function.

Alternatively, example 1 shows a constant string attribute, which can only have been written by the developer, and therefore is assumed to be "safe".

Perhaps the documentation needs to make this clearer.

**Comment by KulvSingh** [comment]
Hi, this is getting compiled now after the solution of having the script function thought event hx-on::after-swap does not seem to get fired.

Thanks.
